java/code/src/com/redhat/rhn/taskomatic/task/MinionActionExecutor.java

```diff
@@ -39,7 +39,8 @@
  */
 public class MinionActionExecutor extends RhnJavaJob {
 
-    private static final int ACTION_DATABASE_GRACE_TIME = 10000;
+    private static final int ACTION_DATABASE_GRACE_TIME = 600_000;
+    private static final int ACTION_DATABASE_POLL_TIME = 100;
```

Comment:
I think we could set the poll time to even 1000 ms per default if we are worried to lookup actions too often, trying once per second should be enough.

Reply:
I did observe problems during mass-onboarding being bottlenecked by this value, as the distribution of waiting times has a long tail:

![waiting_times](https://user-images.githubusercontent.com/250541/46190188-c9e67d00-c2f2-11e8-80e2-54e7a901f1fa.png)

As you can see, most Actions either start immediately or within a few milliseconds. Using a coarser-grained poll time here would hurt importantly. Nevertheless we have a long tail of a few Actions really taking a lot of seconds, so we have to have a pretty big grace time as well :disappointed: 

Note that some 50% of the time spent in RegisterMinionMessageAction is due to addition to system groups - that's some baggage we got from the time this was used for subscription counting. That can definitely be optimized a lot, and I would hope it reduces the variation here as well.

Reply:
Details here https://w3.suse.de/~smoioli/onboarding%20-%20time%20spent%20before%20an%20Action%20is%20available.ods